DevEdu.API/DevEdu.Business.Tests/StudentAnswerOnTaskServiceTests.cs

```diff
@@ -0,0 +1,209 @@
+
+            // When
+            var actualStatusId = sut.ChangeStatusOfStudentAnswerOnTask(taskId, userId, statusId, inputIds);
+
+            // Then
+            Assert.AreEqual(statusId, actualStatusId);
+            _studentAnswerOnTaskRepoMock.Verify(x => x.ChangeStatusOfStudentAnswerOnTask(inputIds), Times.Once);
+        }
+
+        [Test]
+        public void ChangeStatusOfStudentAnswerOnTask_ExistingTaskIdStudentIdAndTaskStatusAcceptedPassed_CompletedDateChanged()
+        {
+            // Given
+            var studentAnswerDto = StudentAnswerOnTaskData.GetStudentAnswerOnTaskDto();
+            var acceptedStatusDto = StudentAnswerOnTaskData.GetStudentAnswerOnTaskWithAcceptedTaskStatusDto();
+            int taskId = StudentAnswerOnTaskData.TaskId;
+            int userId = StudentAnswerOnTaskData.UserId;
+            int statusId = StudentAnswerOnTaskData.AcceptedStatus;
+            var inputIds = StudentAnswerOnTaskData.GetTaskIdStudentIdAndStatusIdDto();
+
+            _studentAnswerOnTaskRepoMock.Setup(x => x.ChangeStatusOfStudentAnswerOnTask(inputIds)).Returns((int)inputIds.TaskStatus);
+            _studentAnswerOnTaskRepoMock.Setup(x => x.GetStudentAnswerOnTaskByTaskIdAndStudentId(taskId, userId)).Returns(acceptedStatusDto);
+
+            var sut = new StudentAnswerOnTaskService(_studentAnswerOnTaskRepoMock.Object);
+
+            // When
+            var actualStatusId = sut.ChangeStatusOfStudentAnswerOnTask(taskId, userId, statusId, inputIds);
+            var dto = sut.GetStudentAnswerOnTaskByTaskIdAndStudentId(taskId, userId);
+
+            // Then
+            Assert.AreEqual(acceptedStatusDto.CompletedDate, dto.CompletedDate);
+            _studentAnswerOnTaskRepoMock.Verify(x => x.ChangeStatusOfStudentAnswerOnTask(inputIds), Times.Once);
+            _studentAnswerOnTaskRepoMock.Verify(x => x.GetStudentAnswerOnTaskByTaskIdAndStudentId(taskId, userId), Times.Once);
+        }
+
+
+        [Test]
+        public void UpdateStudentAnswerOnTask_ExistingTaskIdStudentIdAndTaskAnswerDtoPassed_ReturnStudentAnswerOnTaskDto()
+        {
+            // Given
+            var changedStudentAnswerDto = StudentAnswerOnTaskData.GetChangedStudentAnswerOnTaskDto();
+            var studentAnswerDto = StudentAnswerOnTaskData.GetStudentAnswerOnTaskDto();
+            int taskId = StudentAnswerOnTaskData.TaskId;
+            int userId = StudentAnswerOnTaskData.UserId;
+            var onlyAnswer = StudentAnswerOnTaskData.GetAnswerOfStudent();
+            var inputIds = StudentAnswerOnTaskData.GetTaskIdStudentIdAndStatusIdDto();
+
+            _studentAnswerOnTaskRepoMock.Setup(x => x.GetStudentAnswerOnTaskByTaskIdAndStudentId(taskId, userId)).Returns(studentAnswerDto);
+            _studentAnswerOnTaskRepoMock.Setup(x => x.UpdateStudentAnswerOnTask(onlyAnswer));
+            _studentAnswerOnTaskRepoMock.Setup(x => x.GetStudentAnswerOnTaskByTaskIdAndStudentId(taskId, userId)).Returns(changedStudentAnswerDto);
```

Comment:
very strange setups for GetStudentAnswerOnTaskByTaskIdAndStudentId
first setup says that when it receives some taskId (for example, 42), the method returns a list
and the second says that when it receives the same taskId (42), the method returns another list
how does it work?..